crates/nu-cli/src/completions/completion_common.rs

```diff
@@ -156,16 +156,25 @@ pub fn complete_item(
     engine_state: &EngineState,
     stack: &Stack,
 ) -> Vec<(nu_protocol::Span, String, Option<Style>)> {
-    let partial = surround_remove(partial);
-    let isdir = partial.ends_with(is_separator);
+    let cleaned_partial = surround_remove(partial);
+    let isdir = cleaned_partial.ends_with(is_separator);
+    let expanded_partial = expand_ndots(Path::new(&cleaned_partial));
+    let should_collapse_dots = expanded_partial != Path::new(&cleaned_partial);
```

Comment:
So this means that if I type `cd ../../`, that won't get collapsed, but if I type `cd ../.../`, that'll get collapsed to `..../` rather than `../.../`, right? That behavior makes sense to me - collapse as much possible, but only if we know the user's fine with ndots.